web/impact/impact/v1/helpers/matching_program_criterion_helper.py

```diff
@@ -12,13 +12,21 @@
 class MatchingProgramCriterionHelper(MatchingCriterionHelper):
     application_field = "startup_id"
     judge_field = "expertprofile__home_program_family__name"
+    program_families = None
 
     def __init__(self, subject):
         super().__init__(subject)
         self._program_name_cache = None
 
+    @classmethod
+    def _program_family_map(cls):
+        if cls.program_families is None:
+            cls.program_families = cls.cache_instances_by_name(ProgramFamily)
+        return cls.program_families
+
     def app_ids_for_feedbacks(self, feedbacks, option_name, applications):
-        target = ProgramFamily.objects.filter(name=option_name).first()
+        target = self.__class__._program_family_map()[option_name]
```

Comment:
if you are accessing a class attribute use MatchingProgramCriterionHelper._program_family_map()

Reply:
Sadly, that leads to a ridiculously long line

Reply:
im not seeing your point - you should still be able to use MatchingProgramCriterionHelper._program_family_map() and make it pass pep8

Reply:
```
>>> len("        target = MatchingProgramCriterionHelper._program_family_map()[option_name]")
82
>>> ```

Can you suggest your preferred change? I don't see why `self.__class__` is an issue

Reply:
@pbx i would be interested to see what you think about this.